include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_fpga.h

```diff
@@ -77,218 +77,35 @@ struct __parallel_for_fpga_submitter<__internal::__optional_kernel_name<_Name...
-auto
-__parallel_stable_sort(_ExecutionPolicy&& __exec, _Range&& __rng, _Compare __comp, _Proj __proj)
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_stable_sort(__exec.__device_policy(),
-                                                                     ::std::forward<_Range>(__rng), __comp, __proj);
-}
-
-//------------------------------------------------------------------------
-// parallel_partial_sort
-//-----------------------------------------------------------------------
-
-// TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Iterator, typename _Compare,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
-auto
-__parallel_partial_sort(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __mid, _Iterator __last,
-                        _Compare __comp)
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_partial_sort(__exec.__device_policy(), __first, __mid, __last,
-                                                                      __comp);
-}
-
 //------------------------------------------------------------------------
 // parallel_histogram
 //-----------------------------------------------------------------------
 
 // TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
+template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr>
 auto
-__parallel_histogram(_ExecutionPolicy&& __exec, const _Event& __init_event, _Range1&& __input, _Range2&& __bins,
-                     const _BinHashMgr& __binhash_manager)
+__parallel_histogram(oneapi::dpl::__internal::__fpga_backend_tag, _ExecutionPolicy&& __exec, const _Event& __init_event,
+                     _Range1&& __input, _Range2&& __bins, const _BinHashMgr& __binhash_manager)
 {
-    if constexpr (sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t))
-    {
-        // workaround until we implement more performant version for patterns
-        return oneapi::dpl::__par_backend_hetero::__parallel_histogram(
-            __exec.__device_policy(), __init_event, ::std::forward<_Range1>(__input), ::std::forward<_Range2>(__bins),
-            __binhash_manager);
-    }
-    else
-    {
-        static_assert(false, "histogram is not supported on FPGA devices with output types greater than 32 bits");
-        return __future(sycl::event{});
-    }
+    static_assert(sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t),
```

Comment:
Hi, @danhoeflinger 

is this check
```C++
    static_assert(sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t),
                  "histogram is not supported on FPGA devices with output types greater than 32 bits");
```

really required here or it can be moved into `__parallel_histogram(oneapi::dpl::__internal::__device_backend_tag` implementation?

Reply:
This restriction is specific to FPGA devices, and the intent is to not enforce a restriction on other devices.

We could possibly move this up and combine the check with a check of FPGA policy / tag if that makes things easier for this PR.  I've not studied this PR in detail to understand its impact in combination.

Reply:
So if this restriction is specific to FPGA devices we should save this code as is.
Thanks!